pkg/vrpqstatus/status.go

```diff
@@ -60,18 +61,30 @@ func Update(ctx context.Context, clnt client.Client, log logr.Logger, vrpq *vapi
 	})
 }
 
-// UpdateConditionAndState will update a condition and state status
-// This is a no-op if the status condition is already set.  The input vrpq will
+// UpdateStatusForRestorePointsQuery will update a condition, state and restore points status
+// This is a no-op if the status condition is already set. The input vrpq will
 // be updated with the status condition.
-func UpdateConditionAndState(ctx context.Context, clnt client.Client, log logr.Logger,
-	vrpq *vapi.VerticaRestorePointsQuery, condition *metav1.Condition, state string) error {
-	// refreshConditionInPlace will update the status condition in vrpq.  The update
-	// will be applied in-place.
+func UpdateStatusForRestorePointsQuery(ctx context.Context, clnt client.Client, log logr.Logger,
+	vrpq *vapi.VerticaRestorePointsQuery, conditions []*metav1.Condition, state string,
+	restorePoints []vclusterops.RestorePoint) error {
+	// refreshConditionInPlace will update the status condition, state and
+	// restore points in vrpq.  The update will be applied in-place.
 	refreshConditionInPlace := func(vrpq *vapi.VerticaRestorePointsQuery) error {
 		if vrpq.Status.State != state {
 			vrpq.Status.State = state
 		}
-		meta.SetStatusCondition(&vrpq.Status.Conditions, *condition)
+		for _, condition := range conditions {
+			meta.SetStatusCondition(&vrpq.Status.Conditions, *condition)
+		}
+		if len(vrpq.Status.RestorePoints) < len(restorePoints) {
+			vrpq.Status.RestorePoints = make([]vapi.RestorePoint, len(restorePoints))
+		}
+		for i := range restorePoints {
+			vrpq.Status.RestorePoints[i].Archive = restorePoints[i].Archive
+			vrpq.Status.RestorePoints[i].ID = restorePoints[i].ID
+			vrpq.Status.RestorePoints[i].Index = restorePoints[i].Index
+			vrpq.Status.RestorePoints[i].Timestamp = restorePoints[i].Timestamp
+		}
 		return nil
 	}
 	return Update(ctx, clnt, log, vrpq, refreshConditionInPlace)
```

Comment:
Do we have any use cases of callers external to this package calling Update? I wonder if UpdateStatusForRestorePointsQuery should be renamed to Update, and Update renamed to updateImpl.